### PR TITLE
Analytics: Add Matomo script

### PR DESCRIPTION
**Pull request by masonmcelvain**
Tracking code from: https://minerva.ifixit.com/index.php?module=CoreAdminHome&action=trackingCodeGenerator

## QA

Click around on dev and prod previews, and make sure the event show up in [matomo.ubreakit.com](http://matomo.ubreakit.com/) or [minerva.ifixit.com](https://minerva.ifixit.com). Thanks!

_Note:_ You should disable all ad blockers when testing this, as they might prevent the Matomo script from running.

CC @sterlinghirsh

Connects #300

### CODE REVIEW

**Comment by masonmcelvain** [comment]
dev_block ✌🏻

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻 Not sure how to test these previews, because they seem to be trying to talk to minerva.ifixit.com. @jarstelfox maybe you have some insight here?

image

**Comment by sterlinghirsh** [comment]
CR :zap: FYI I also found this package https://github.com/SocialGouv/matomo-next. Not sure if it's necessary or not, but we'll have to validate that in QA. The preview should be talking to matomo.ubreakit.com right?

**Comment by masonmcelvain** [comment]
Cool package, thanks for finding! Happy to install it if we think it's simpler, looks like it's only 37 kb.

> The preview should be talking to matomo.ubreakit.com right?

It depends which env vars the previews are using. If they use the Prod env vars, they'll try to talk to minerva, not ubreakit.

**Comment by masonmcelvain** [comment]
> Not sure how to test these previews

I also forgot that you need to disable ad blockers when testing analytics, I'll add that to QA steps

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Both previews are reporting to minerva.ifixit.com, but I didn't see any events show up. I took the POST request to the tracker API and modified it to send to matomo.ubreakit.com so that I could find the event a little more easily, but still wasn't able to find the event.

**Comment by jarstelfox** [comment]
> Not sure how to test these previews, because they seem to be trying to talk to minerva.ifixit.com.

I found this on https://github.com/iFixit/react-commerce/pull/459

It looks like the E2E Tests / vercel are using the production `.env`. I made a small change for the `NODE_ENV` so that tests were their own environment

**Review by djmetzle — APPROVED** [review]
Nice! CR :footprints:

**Comment by jarstelfox** [comment]
QA 🌵 

I ran a prod build locally and received the prod URL:

image

Running the dev build produces `https://matomo.ubreakit.com`

The vercel preview shows the correct URL:

image

File: frontend/components/analytics/Matomo.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                  var _paq = window._paq = window._paq || [];
+                  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
+                  _paq.push(['trackPageView']);
+                  _paq.push(['enableLinkTracking']);
+                  (function() {
+                     var u='${MATOMO_URL}';
+                     _paq.push(['setTrackerUrl', u+'/js/tracker.php']);
+                     _paq.push(['setSiteId', '1']);
+                     var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+                     g.async=true; g.src=u+'/js/tracker.php'; s.parentNode.insertBefore(g,s);
```
Comment (djmetzle) [inline frontend/components/analytics/Matomo.tsx:20]: Quick check:

In production, we lightly obfuscate the tracker path (to work around basic adblockers):
https://github.com/iFixit/ifixit/blob/95a280d5a046342b69196ae04f6c082643ef3200/iFixit/Service/Matomo/MatomoPageComponent.php#L72

Comment (djmetzle) [reply frontend/components/analytics/Matomo.tsx:20]: @jarstelfox i think we made `minerva.php` an alias, and `tracker.js` should still work. But do we want to keep config parity there? :thinking:

**Comment by masonmcelvain** [comment]
@jarstelfox That QA makes sense to me! We want to see `minerva` when using prod env vars, and `matomo` when we use dev env vars, and it looks like you saw that.

File: frontend/components/analytics/Matomo.tsx (hunk 1 above)
Comment (jarstelfox) [reply frontend/components/analytics/Matomo.tsx:20]: Yea that would be nice to add, I can re-qa if needed

Comment (masonmcelvain) [reply frontend/components/analytics/Matomo.tsx:20]: Can do. Will `matomo.ubreakit.com/minerva.php` work? Or should this obfuscation only be for `minerva.ifixit.com`?

Comment (djmetzle) [reply frontend/components/analytics/Matomo.tsx:20]: Both work, right?
https://matomo.ubreakit.com/minerva.php

Comment (masonmcelvain) [reply frontend/components/analytics/Matomo.tsx]: Didn't even think to follow the link myself 😅 thanks!

**Comment by masonmcelvain** [comment]
un_deploy_block ✌🏻

**Comment by jarstelfox** [comment]
QA 🌵 

image

**Review by jarstelfox — APPROVED** [review]
CR 🌵

**Pull request merged**